### ISSUE TITLE
Any suggestion on Whole Body RL?

### ISSUE THREAD

**Issue by zitongbai** [unitreerobotics/unitree_rl_gym#49]
Hi there. Thanks for sharing your work. 

I noticed that in G1 training config, the action space only includes 12 leg dof, while the upper dofs are ignored and set fixed in urdf. I am trying to use the whole body model `g1_29dof_lock_waist_rev_1_0` and use all the dofs (including both leg and upper dofs) in the action space and observation space. 

However, it is extremely challenging for the training process to converge to the global optimum, as it often gets stuck in local optima. 

Here is the reward function I added for the upper dofs: 

```python
    def _reward_arm_dof_deviation(self):
        diff = self.dof_pos[:, self.arm_dof_indices] - self.default_dof_pos[:,self.arm_dof_indices]
        return torch.sum(torch.square(diff), dim=1)
    
    def _reward_waist_dof_deviation(self):
        diff = self.dof_pos[:, self.waist_dof_indices] - self.default_dof_pos[:,self.waist_dof_indices]
        return torch.sum(torch.square(diff), dim=1)
```

where `self.arm_dof_indices` and `self.waist_dof_indices` are defined inside `legged_robot.py`'s `_create_envs`: 
```python
        self.arm_dof_indices = []
        self.waist_dof_indices = []
        for i in range(len(self.dof_names)):
            if "shoulder" in self.dof_names[i]:
                self.arm_dof_indices.append(i)
            if "elbow" in self.dof_names[i]:
                self.arm_dof_indices.append(i)
            if "wrist" in self.dof_names[i]:
                self.arm_dof_indices.append(i)
            
            if "waist" in self.dof_names[i]:
                self.waist_dof_indices.append(i)
                
        self.arm_dof_indices = torch.tensor(self.arm_dof_indices, dtype=torch.long, device=self.device, requires_grad=False)
        self.waist_dof_indices = torch.tensor(self.waist_dof_indices, dtype=torch.long, device=self.device, requires_grad=False)
```

the corresponding reward scales are: 

```python
            arm_dof_deviation = -0.5
            waist_dof_deviation = -0.1
```

Could you please share some suggestions on how to train the whole body RL based on your repo? 

Thanks for your attention in advance.

**Comment by JoeZhang1116** [comment]
Hi, great to see your work! I am also doing RL training on G1 these days. The urdf file I am using is g1_29dof.urdf with the xml file g1_29dof.xml. I modified parameters like num_observations and num_actions, initialized all the new joints and added a new reward function to keep the upper body stable. But the training result turned out to be disastrous, with the G1 flying  around in the gym. Also,  the training data feedback is abnormal, with a relatively low reward value and NaN result in function loss and surrogate loss. Any suggestions would be helpful! A lot thanks!

![Image](https://github.com/user-attachments/assets/c887a18d-03b4-4b7e-8cbb-4c3c8550aab3)

https://github.com/user-attachments/assets/72bbc1c0-e37c-4459-ac8c-d9361b4c5379

**Comment by zitongbai** [comment]
I have tried this settings of rewards and it works:

https://github.com/zitongbai/legged_gym/blob/bd7f22cb81ce26dd22542a0bfe47f4061623a0f1/legged_gym/envs/g1/g1_config.py#L125-L146

**Comment by heartInsert** [comment]
@JoeZhang1116   Have you  solve this  problem ?

**Comment by heartInsert** [comment]
@zitongbai 兄弟，你的G1全身控制能跑了吗？我最近也遇到这个问题了
